ServingRemoteData.adoc

```diff
@@ -0,0 +1,239 @@
+= Serving Remote Data, Including Data in S3, with HDF5/NetCDF-4 and DMR++
+:Nathan Potter <ndp@opendap.org>:
+{docdate}
+:numbered:
+:toc:
+
+== Overview
+There are several ways that Hyrax can serve remotely located data. In this paper we discuss how to construct a Hyrax data system that utilizes Amazon's S3 service (or any HTTP service that supports the use of Range GET) in conjunction with local metadata held in `dmr++` files to create a system that provides quick access to data and that minimizes (for constrained DAP requests) the data egress volume from the HTTP service.
+
+In this scheme the original data files are held in S3 (or any other HTTP service that supports Range GET) as objects. The metadata, along with access information about the S3 object, are held as `dmr++` files on the Hyrax platform.
+
+Each `dmr\++` file needs to know the network location(s) of the objects(s) from which its data will be read. For some representations multiple objects may be used, but for these examples each `dmr++` will be associated with a single web accessible object (held in S3 for this example).
+
+=== Process
+
+Most of this process is about creating and organizing content for Hyrax.
+
+1. Process the data to create the content - `dmr++` files
+1. Place the `dmr++` files into a Hyrax server's data tree.
+1. Making sure the S3 bucket URL is on the whitelist.
```

Comment:
I believe that we will need to add credentials to the server so that it can access the S3 bucket unless that bucket is open to the world _or_ the server is running on an EC2 in the same account as the S3 bucket. And, even then, maybe not. (What I said regarding the OPeNDAP account and being able to access S3 without auth might have been wrong; I might have have the ID and Secret in a cred config file).

Reply:
But feel free to merge this and we can see how it all looks. Maybe a test drive would be a good idea?

Reply:
yes - a test drive sounds good. @ndp-opendap want to merge it and see how it looks like?